istresearch/ansible-symphony
# PULSE-4738: Implement DataDog Metrics Collection in Storm

Added library and tasks for Storm metrics monitoring. Default metrics for each topology will be sent to a statsd daemon.

## Variables

Variables introduced: /roles/storm

* storm_lib_dir
* storm_metrics
* storm_metrics_statsd_host
* storm_metrics_statsd_port
* storm_metrics_statsd_prefix

Variables modified:

* storm_topology_extra_config

## Component configuration failure conditions

* Incorrect statsd daemon settings

## Ansible deployment command

    `$ ansible-playbook -u <user> -i <system>.inventory --tags deps-storm site-infrastructure.yml`

## Additional deployment steps

* Ensure host var has the correct settings for the new variables prior to deployment.
* In order to actually confirm that Storm is sending metrics correctly to the defined statsd daemon, you will need to submit a topology.

## Steps to verify deployment

1. After the site-infrastructure playbook has completed, run `supervisorctl status` to confirm all required components are running:
```
storm-logviewer                 RUNNING    pid 7905, uptime 3:48:33
storm-nimbus                    RUNNING     pid 5130, uptime 3:49:24
storm-supervisor               RUNNING     pid 7738, uptime 3:48:43
storm-ui                             RUNNING    pid 6422, uptime 3:49:04
zookeeper                          RUNNING    pid 30999, uptime 3:50:45
```
2. After submitting a topology, you should see a bolt for the `StatsdMetricConsumer`. You can view this in the Storm UI when clicking on the test topology.
**NOTE:** You may have to click the **Show System Stats** button on the bottom of the page for this to be visible.

![storm-metrics-ui](https://user-images.githubusercontent.com/1586713/31081391-bc22abc2-a759-11e7-8e73-ef40ec013197.png)

3. After about a minute after you have submitted the topology, the statsd daemon should be receiving the metrics. Check the daemon to confirm the metrics are being sent. The default prefix for the metrics is `storm.metrics.`.

![storm-metrics-graphite](https://user-images.githubusercontent.com/1586713/31081404-c7c28f1a-a759-11e7-86d4-5fa5a11d3f50.png)
